src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -262,9 +262,14 @@ def __getattr__(self, key: str) -> Any:
     def _binary_ops(self, method: str, other: Any) -> Index:
         if isinstance(other, Index):
             other = other.to_series().reset_index(drop=True)
-        return self.__constructor__(
-            self.to_series().reset_index(drop=True).__getattr__(method)(other)
-        )
+        series = self.to_series().reset_index(drop=True).__getattr__(method)(other)
+        qc = series._query_compiler
+        qc = qc.set_index_from_columns(qc.columns, include_index=False)
+        # Use base constructor to ensure that the correct type is returned.
+        idx = Index(query_compiler=qc)
+        if series.name is None:
+            idx.name = None
```

Comment:
Why not do: `idx = Index(query_compiler=qc, name=series.name)`?

Reply:
I thinks that should also work, let me check.

Reply:
Oh, right now we don't support `name` argument in lazy index constructor.

Reply:
Maybe we can set it as `idx.name = series.name`?

Reply:
that worked!